src/chainstate/stacks/db/mod.rs

```diff
@@ -762,15 +773,30 @@ impl StacksChainState {
                 ],
             )?;
 
-            StacksChainState::apply_schema_migrations(&tx, mainnet, chain_id)?;
+            if migrate {
+                StacksChainState::apply_schema_migrations(&tx, mainnet, chain_id)?;
+            }
         }
 
         dbtx.instantiate_index()?;
         dbtx.commit()?;
         Ok(marf)
     }
 
-    fn load_db_config(conn: &DBConn) -> Result<DBConfig, Error> {
+    /// Load the chainstate DBConfig, given the path to the chainstate root
+    pub fn get_db_config_from_path(chainstate_root_path: &str) -> Result<DBConfig, db_error> {
+        let index_pathbuf =
+            StacksChainState::header_index_root_path(PathBuf::from(chainstate_root_path));
+        let index_path = index_pathbuf
+            .to_str()
+            .ok_or_else(|| db_error::ParseError)?
+            .to_string();
+
+        let marf = StacksChainState::open_index(&index_path)?;
+        StacksChainState::load_db_config(marf.sqlite_conn())
+    }
```

Comment:
This is only ever used in the new test, right? Can you move it to `coordinator::tests`?

Reply:
Nevermind -- I see it gets used in the schema checks.